test/common/upstream/health_checker_impl_test.cc

```diff
@@ -1452,6 +1494,860 @@ TEST_F(RedisHealthCheckerImplTest, AllDontReuseConnection) {
+    }
+    if (!trailers_empty) {
+      auto trailers = std::make_unique<Http::TestHeaderMapImpl>();
+      for (const auto& header : spec.trailers) {
+        trailers->addCopy(header.first, header.second);
+      }
+      test_sessions_[index]->stream_response_callbacks_->decodeTrailers(std::move(trailers));
+    }
+  }
+
+  MOCK_METHOD2(onHostStatus, void(HostSharedPtr host, bool changed_state));
+
+  std::shared_ptr<MockCluster> cluster_;
+  NiceMock<Event::MockDispatcher> dispatcher_;
+  std::vector<TestSessionPtr> test_sessions_;
+  std::shared_ptr<TestGrpcHealthCheckerImpl> health_checker_;
+  NiceMock<Runtime::MockLoader> runtime_;
+  NiceMock<Runtime::MockRandomGenerator> random_;
+  std::list<uint32_t> connection_index_{};
+  std::list<uint32_t> codec_index_{};
+};
+
+class GrpcHealthCheckerImplTest : public GrpcHealthCheckerImplTestBase, public testing::Test {};
+
+// Most of these test are the same as HttpHealthCheckerImplTest.
+TEST_F(GrpcHealthCheckerImplTest, Success) {
+  setupServiceNameHC();
+  EXPECT_CALL(*this, onHostStatus(_, false)).Times(1);
+
+  cluster_->prioritySet().getMockHostSet(0)->hosts_ = {
+      makeTestHost(cluster_->info_, "tcp://127.0.0.1:80")};
+  cluster_->info_->stats().upstream_cx_total_.inc();
+  expectSessionCreate();
+  expectStreamCreate(0);
+  EXPECT_CALL(*test_sessions_[0]->timeout_timer_, enableTimer(_));
+  EXPECT_CALL(test_sessions_[0]->request_encoder_, encodeHeaders(_, false))
+      .WillOnce(Invoke([&](const Http::HeaderMap& headers, bool) {
+        EXPECT_EQ(Http::Headers::get().ContentTypeValues.Grpc,
+                  headers.ContentType()->value().c_str());
+        EXPECT_EQ(std::string("/grpc.health.v1.Health/Check"), headers.Path()->value().c_str());
+      }));
+  EXPECT_CALL(test_sessions_[0]->request_encoder_, encodeData(_, true))
+      .WillOnce(Invoke([&](Buffer::Instance& data, bool) {
+        std::vector<Grpc::Frame> decoded_frames;
+        Grpc::Decoder decoder;
+        ASSERT_TRUE(decoder.decode(data, decoded_frames));
+        ASSERT_EQ(1U, decoded_frames.size());
+        auto& frame = decoded_frames[0];
+        Buffer::ZeroCopyInputStreamImpl stream(std::move(frame.data_));
+        grpc::health::v1::HealthCheckRequest request;
+        ASSERT_TRUE(request.ParseFromZeroCopyStream(&stream));
+        EXPECT_EQ("service", request.service());
+      }));
+  health_checker_->start();
+
+  EXPECT_CALL(runtime_.snapshot_, getInteger("health_check.max_interval", _));
+  EXPECT_CALL(runtime_.snapshot_, getInteger("health_check.min_interval", _))
+      .WillOnce(Return(45000));
+  EXPECT_CALL(*test_sessions_[0]->interval_timer_, enableTimer(std::chrono::milliseconds(45000)));
+  EXPECT_CALL(*test_sessions_[0]->timeout_timer_, disableTimer());
+  respondServiceStatus(0, grpc::health::v1::HealthCheckResponse::SERVING);
+  EXPECT_TRUE(cluster_->prioritySet().getMockHostSet(0)->hosts_[0]->healthy());
+}
+
+TEST_F(GrpcHealthCheckerImplTest, SuccessResponseSplitBetweenChunks) {
```

Comment:
Generally prefer a single line `//` comment above each test explaining in plain language what the test is trying to accomplish. E.g. `// Validate that a regular response split across chunks is handled.`

Reply:
done